### PR TITLE
CI: Use a github workflow to run west command tests.

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by galak** [comment]
@mbolivar I'm not familiar with pytest - but this shows we fail the bossac test on macos x.  Is there a way to filter based on platform?

**Comment by mbolivar** [comment]
> @mbolivar I'm not familiar with pytest - but this shows we fail the bossac test on macos x. Is there a way to filter based on platform?

Yes, although I would prefer we not do that. Let me look at the failure.

**Comment by mbolivar** [comment]
NVM, let's just skip it. I see what you mean. You can cherry pick https://github.com/mbolivar/zephyr/tree/gh-workflow-west-tests.

**Review by mbolivar — APPROVED** [review]

**Comment by galak** [comment]
> NVM, let's just skip it. I see what you mean. You can cherry pick https://github.com/mbolivar/zephyr/tree/gh-workflow-west-tests.

thanks, pulled it in.

**Comment by galak** [comment]
@mbolivar on thoughts on PYTHONPATH and windows?

**Comment by mbolivar** [comment]
> @mbolivar on thoughts on PYTHONPATH and windows?

scripts/west_commands/README.txt says:

```
Windows users will need to find the path to .west/west/src in their
Zephyr installation, then run something like this:

> cmd /C "set PYTHONPATH=path\to\zephyr\scripts\west_commands && py.test"
```

That worked the last time I tested it, at least.

File: .github/workflows/west_cmds.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        key: ${{ runner.os }}-pip-${{ matrix.python-version }}
+        restore-keys: |
+          ${{ runner.os }}-pip-${{ matrix.python-version }}
+    - name: install pytest
+      run: |
+        pip3 install pytest west
+
+    - name: run pytest
+      run: |
+        PYTHONPATH=./scripts/west_commands pytest ./scripts/west_commands/tests/
```
Comment (mbolivar) [inline .github/workflows/west_cmds.yml]: Is this something you can customize on a per-platform basis, for instance?

Comment (galak) [reply .github/workflows/west_cmds.yml]: I'll look and see if I can.

**Review by nashif — CHANGES_REQUESTED** [review]
cool

**Review by nashif — APPROVED** [review]

**Pull request merged**